modules/navier_stokes/doc/content/source/fvkernels/PINSFVMomentumPressure.md

```diff
@@ -0,0 +1,14 @@
+# PINSFVMomentumPressure
+
+This object adds the $\epsilon \nabla p$ term of the
+incompressible Navier Stokes momentum equation as a volumetric term.
+This formulation causes oscillation near porosity discontinuities and the
+[PINSFVMomentumPressureFlux.md] kernel should be preferred then.
```

Comment:
When you run with the flux formulation do you also include the second term that comes out of the integration by parts? E.g. the p * grad_eps? If not, then I wonder if that's the reason that the flux version "performs better"

Reply:
Not without friction, or if the friction term is the same on both sides (which it would be)

It's definitely the reason it performs better. We're getting rid of the gradient near a discontinuity by doing this. The pressure or the porosity gradient both don't help

Reply:
Hmm I feel like the result will be wrong if we totally disregard the gradient (either pressure or porosity)

Reply:
I ll run the smooth case again but I think the result was closer without the gradient. That, the genFoam paper getting rid of the advection porosity gradient terms and the lack of oscillations (in some cases) were the main reasons I removed it

Reply:
Pressure drop should be correct with this div(eps * P) formulation, since by definition it conserves that term

Reply:
It definitely conserves eps * P, but that's not the term in the governing equation nor for example does div(eps*P) (by itself) yield the correct physical results for compressible fluids. But it does for incompressible fluids then that would be marvelous